extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -0,0 +1,344 @@
+   * the element referenced by the `id` argument.
+   * Examples:
+   *  // Opens the element tapped.
+   *  on="tap:myLightboxViewer'
+   *
+   *  // Opens the element referenced by elementId
+   *  on="tap:myLightboxViewer.open(id='<elementId>')
+   * @override
+   */
+  activate(invocation) {
+    let target = invocation.source;
+    if (invocation.args && invocation.args.id) {
+      const targetId = invocation.args.id;
+      target = this.win.document.getElementById(targetId);
+      user().assert(target,
+        'amp-lightbox-viewer.open: element with id: %s not found', targetId);
+    }
+    this.open_(target);
+  }
+
+  /**
+   * Opens the lightbox-viewer and displays the given element inside.
+   * @param {!Element} elem Element to lightbox.
+   * @private
+   */
+  open_(elem) {
+    if (this.activeElem_ == elem) {
+      return;
+    }
+
+    this.updateViewer_(elem);
+
+    this.element.style.display = 'block';
+    this.active_ = true;
+
+    this.win.document.documentElement.addEventListener(
+        'keydown', this.boundHandleKeyboardEvents_);
+  }
+
+  /**
+   * Closes the lightbox-viewer
+   * @private
+   */
+  close_() {
+    if (!this.active_) {
+      return;
+    }
+
+    this.element.style.display = 'none';
+    this.tearDownElem_(this.activeElem_);
+
+    this.activeElem_ = null;
+    this.active_ = false;
+
+    this.win.document.documentElement.removeEventListener(
+        'keydown', this.boundHandleKeyboardEvents_);
+  }
+
+  /**
+   * Opens the next element to be displayed in the lightbox.
+   * @private
+   */
+  next_() {
+    dev().assert(this.activeElem_);
+    this.manager_.getNext(this.activeElem_).then(nextElem => {
+      if (nextElem) {
+        this.updateViewer_(nextElem);
+      }
+    });
+  }
+
+  /**
+   * Opens the previous element to be displayed in the lightbox.
+   * @private
+   */
+  previous_() {
+    dev().assert(this.activeElem_);
+    this.manager_.getPrevious(this.activeElem_).then(prevElem => {
+      if (prevElem) {
+        this.updateViewer_(prevElem);
+      }
+    });
+  }
+
+  /**
+   * Updates the viewer to display the new element and tears down the old one
+   * @param {!Element} newElem
+   * @private
+   */
+  updateViewer_(newElem) {
+    const previousElem = this.activeElem_;
+    dev().assert(newElem);
+    dev().assert(newElem != previousElem);
+
+    // tear down the previous element
+    if (previousElem) {
+      this.tearDownElem_(previousElem);
+    }
+
+    // setup the new element
+    this.setupElem_(newElem);
+
+    // update active element to be the new element
+    this.activeElem_ = newElem;
+
+    // update the controls
+    this.updateControls_();
+
+    // TODO(aghassemi): Preloading of +/- 1 elements
+
+    // TODO(aghassemi): This is a giant hack.
+    // Find a proper way of scheduling layout for a resource that does not
+    // not belong to the element requesting the layout.
+    if (newElem.resources_) {
```

Comment:
This is fine as a hack, but to be clear: this will not work in minified binaries most likely. The extensions are minified independently from core and we obfuscate all properties trailing with `_`. On the other hand, just checking for `newElem['__AMP__RESOURCE']` would still work. So you can use that.

For my info, `scheduleLayout`, `scheduleInViewport` will not work here because the element is not in the subtree of the lightbox, right? That restriction might be overly paranoid, but that's for another PR.